pkg/util/push/push.go

```diff
@@ -73,9 +75,14 @@ func handler(maxRecvMsgSize int,
 	allowSkipLabelNameValidation bool,
 	push Func,
 	parser parserFunc,
+	haReplicaChecker distributor.HaShortcutRequestCheckerFunc,
 ) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		ctx := r.Context()
+		if haReplicaChecker(ctx, w, r) {
```

Comment:
If we do this:

```suggestion
		if shortcut != && shortcut(ctx, w, r) {
```

We don't need the noop things everywhere, and we avoid unnecessary call in the ingesters.